src/App.vue

```diff
@@ -315,6 +324,12 @@ export default {
     WatchersPopup,
   },
   watch: {
+    previewData: {
+      deep: true,
```

Comment:
do we really need to watch every single property?

Reply:
We need to react to any change and refresh the preview data, but I will review a better way to do it.

Reply:
Removed the watchers, and using events to refresh the preview data